test/examples/examples.spec.js

```diff
@@ -0,0 +1,168 @@
+'use strict';
+
+let _ = require('underscore');
+let Promise = require('bluebird');
+let chakram = require('chakram');
+var expect = chakram.expect;
+let execAsync = Promise.promisify(require('child_process').exec);
+let fs = require('fs');
+let http = require('http');
+let logger = require('mocha-logger');
+let path = require('path');
+let retry = require('bluebird-retry');
+let spawn = require('child_process').spawn;
+let WebSocket = require('ws');
+
+function spawnify(command, args, options) {
+    return new Promise((resolve, reject) => {
+        args = args || [];
+        logger.log('spawning "' + command + ' ' + args.join(' '));
+        options = _.extend( {
+            detached: true,
+            stdio: ['ignore', 'pipe', 'pipe']
+        }, options);
+
+        var spawned = spawn(command, args, options);
+        var stdout = '';
+        var stderr = '';
+
+        spawned.stdout.on('data', (data) => {
+            logger.log('STDOUT:', data);
+            stdout += data;
+        });
+
+        spawned.stderr.on('data', (data) => {
+            logger.log('STDERR:', data);
+            stderr += data;
+        });
+
+        spawned.on('close', (code) => {
+            if (code === 0) {
+                resolve(stdout, stderr);
+            } else {
+                reject(Error('command: "' + command + ' ' + args.join(' ') +
+                             '", failed with ' + code));
+            }
+        });
+    });
+}
+
+describe('examples', () => {
+    let hostAddress = 'localhost';
+    let juttleEnginePort = 8080;
+    let baseUrl = `http://${hostAddress}:${juttleEnginePort}`
+    let baseAPIUrl = `${baseUrl}/api/v0`
+
+    let retryHTTPOptions = {
+        interval: 500,
+        timeout: 10000
+    }
+
+    before(() => {
+        return execAsync('docker run ubuntu route | grep default | awk \'{print $2}\'')
```

Comment:
docker-machine would have to be installed in this case (some additional scripts steps in travis.yml) since docker-machine isn't really a requirement for anything else I think sticking with the 1 way that does figure out the gateway without additional dependencies should be fine.